File: base/src/main/java/me/shetj/base/base/IPresenter.java
package me.shetj.base.base;

import androidx.annotation.Keep;

/**
 * @author shetj
 */
@Keep
public interface IPresenter {

    /**
     * 做一些初始化操作
     */
    void onStart();

    /**
     *  Activity#onDestroy() 调用{@link IPresenter#onDestroy()}
     */
    void onDestroy();

}

File: base/src/main/java/me/shetj/base/mvp/IPresenter.kt
package me.shetj.base.mvp

import androidx.annotation.Keep

/**
 * @author shetj
 */
@Keep
interface IPresenter {

    /**
     * 做一些初始化操作
     */
    fun onStart()

    /**
     * Activity#onDestroy() 调用[IPresenter.onDestroy]
     */
    fun onDestroy()

}